src/Form/formatters.js

```diff
@@ -163,14 +163,13 @@ const getMMAndYYYYFromString = (expiration) => {
  * @param {InternalDataStorageObject} credentials
  * @return {boolean}
  */
-const shouldStoreCredentials = ({ credentials }) => Boolean(credentials.password);
-
-/**
- * @param {InternalDataStorageObject} credentials
- * @return {boolean}
- */
-const shouldStoreIdentities = ({ identities }) =>
-    Boolean((identities.firstName || identities.fullName) && identities.addressStreet && identities.addressCity);
+const shouldStoreIdentities = ({ identities }) => {
+    return (
+        Boolean((identities.firstName || identities.fullName) && identities.addressStreet && identities.addressCity) ||
+        Boolean(identities.emailAddress) ||
+        Boolean(identities.phone)
```

Comment:
Will this start prompting to save identities that just have a lone phone or lone email? Do we have some backend checks in the native apps to prevent that?

Reply:
Ah, we have a guard against it in the interface prototype. Got it.

But that makes this whole thing confusing because we're saying that we should store identities and we shouldn't.

What I think we should do instead is: when we detect one of these email-only or phone-only forms, move that email or phone to `credentials.username`, then let the rest of the flow continue as usual. Because that's actually what we want, logically. It's not an identity, it's a username. Makes sense? I think this should also simplify things on the native side, because they can only expect these updated calls to have `credentials.username` rather than multiple potential types.